challenge_7/python/tvanderwal/src/findMissingNumber.py

```diff
@@ -1,12 +1,21 @@
 #Challenge 7: Return a missing integer from an unsort list of integers
-
+"""
+old solution. intList.sort() is O(nlogn) time and doesn't fit the challenge criteria
 def findMissingNumber(intList):
     #sort list
     intList.sort()
 
     for i, val in enumerate(intList):
         if i != val:
             return i
+"""
+
+def findMissingNumber(intList):
+    sumInts = sum(intList)
+    sumRange = 0
+    for i in range(1, len(intList) + 1):
```

Comment:
The sum from 1 to n is n * (n+1) / 2, so you don't need to loop through from 1 to n, but your method is still O(n).